HUDI merge strategy failing with `File not present on S3`

### Describe the bug
A clear and concise description of what the bug is. What command did you run? What happened?

I am running merge strategy for a very basic model - with a composite unique_key (combination of two columns) - no partitions

I see that the old parquet file is getting deleted and then the stacktrace shows that it was trying to download the deleted file.

How to use `merge` strategy for HUDI incremental models with a composite unique_key (non-partitioned)?

### Steps To Reproduce
The initial model creation works perfect! Only re-running the incremental update fails!

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.
```
18:48:28  Database Error in model <REDACTED> (models/../##.sql)
18:48:28    Glue cursor returned `error` for statement None for code SqlWrapper2.execute('''/* {"app": "dbt", "dbt_version": "1.3.2", "profile_name": "<REDACTED>", "target_name": "dev", "node_id": "<REDACTED>"} */
18:48:28    select * from <REDACTED> limit 1 '''), Py4JJavaError: An error occurred while calling o138.count.
18:48:28    : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 14.0 failed 4 times, most recent failure: Lost task 0.3 in stage 14.0 (TID 18) (172.36.239.234 executor 1): org.apache.spark.sql.execution.datasources.FileDownloadException: Failed to download file path: s3://../../8ca16913-e880-4146-bf9f-4ef66ce87ce5-0_0-8-0_20230203184613191.parquet, range: 0-436972, partition values: [empty row], isDataPresent: false, eTag: 1c025d8bca18dae2fe329749e1e81b17
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader.next(AsyncFileDownloader.scala:142)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.getNextFile(FileScanRDD.scala:279)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:201)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:159)
18:48:28        at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:606)
18:48:28        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
18:48:28        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
18:48:28        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
18:48:28        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:755)
18:48:28        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:454)
18:48:28        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:133)
18:48:28        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
18:48:28        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
18:48:28        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
18:48:28        at org.apache.spark.scheduler.Task.run(Task.scala:131)
18:48:28        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
18:48:28        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
18:48:28        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
18:48:28        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
18:48:28        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
18:48:28        at java.lang.Thread.run(Thread.java:750)
18:48:28    Caused by: java.io.FileNotFoundException: File not present on S3
18:48:28        at com.amazon.ws.emr.hadoop.fs.s3.S3FSInputStream.read(S3FSInputStream.java:273)
18:48:28        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
18:48:28        at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
18:48:28        at java.io.FilterInputStream.read(FilterInputStream.java:83)
18:48:28        at org.apache.parquet.io.DelegatingSeekableInputStream.read(DelegatingSeekableInputStream.java:61)
18:48:28        at org.apache.parquet.bytes.BytesUtils.readIntLittleEndian(BytesUtils.java:80)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:528)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:513)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:507)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:456)
18:48:28        at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat.getFooter(ParquetFileFormat.scala:644)
18:48:28        at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat.$anonfun$buildPrefetcherWithPartitionValues$1(ParquetFileFormat.scala:515)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader.org$apache$spark$sql$execution$datasources$AsyncFileDownloader$$downloadFile(AsyncFileDownloader.scala:93)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader$$anon$1.call(AsyncFileDownloader.scala:73)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader$$anon$1.call(AsyncFileDownloader.scala:72)
18:48:28        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
18:48:28        ... 3 more
18:48:28    
18:48:28    Driver stacktrace:
18:48:28        at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2465)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2414)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2413)
18:48:28        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:58)
18:48:28        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:51)
18:48:28        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2413)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1124)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1124)
18:48:28        at scala.Option.foreach(Option.scala:257)
18:48:28        at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1124)
18:48:28        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2679)
18:48:28        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2621)
18:48:28        at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2610)
18:48:28        at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.checkNoFailures(AdaptiveExecutor.scala:147)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.doRun(AdaptiveExecutor.scala:88)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.tryRunningAndGetFuture(AdaptiveExecutor.scala:66)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveExecutor.execute(AdaptiveExecutor.scala:57)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.$anonfun$getFinalPhysicalPlan$1(AdaptiveSparkPlanExec.scala:184)
18:48:28        at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.getFinalPhysicalPlan(AdaptiveSparkPlanExec.scala:183)
18:48:28        at org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanExec.executeCollect(AdaptiveSparkPlanExec.scala:404)
18:48:28        at org.apache.spark.sql.Dataset.$anonfun$count$1(Dataset.scala:3046)
18:48:28        at org.apache.spark.sql.Dataset.$anonfun$count$1$adapted(Dataset.scala:3045)
18:48:28        at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3724)
18:48:28        at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:107)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:232)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.executeQuery$1(SQLExecution.scala:110)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:135)
18:48:28        at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:107)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.withTracker(SQLExecution.scala:232)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:135)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:253)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:134)
18:48:28        at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:772)
18:48:28        at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:68)
18:48:28        at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3722)
18:48:28        at org.apache.spark.sql.Dataset.count(Dataset.scala:3045)
18:48:28        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:48:28        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:48:28        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:48:28        at java.lang.reflect.Method.invoke(Method.java:498)
18:48:28        at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
18:48:28        at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
18:48:28        at py4j.Gateway.invoke(Gateway.java:282)
18:48:28        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
18:48:28        at py4j.commands.CallCommand.execute(CallCommand.java:79)
18:48:28        at py4j.GatewayConnection.run(GatewayConnection.java:238)
18:48:28        at java.lang.Thread.run(Thread.java:750)
18:48:28    Caused by: org.apache.spark.sql.execution.datasources.FileDownloadException: Failed to download file path: s3://../../8ca16913-e880-4146-bf9f-4ef66ce87ce5-0_0-8-0_20230203184613191.parquet, range: 0-436972, partition values: [empty row], isDataPresent: false, eTag: 1c025d8bca18dae2fe329749e1e81b17
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader.next(AsyncFileDownloader.scala:142)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.getNextFile(FileScanRDD.scala:279)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:201)
18:48:28        at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:159)
18:48:28        at org.apache.spark.sql.execution.FileSourceScanExec$$anon$1.hasNext(DataSourceScanExec.scala:606)
18:48:28        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.columnartorow_nextBatch_0$(Unknown Source)
18:48:28        at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
18:48:28        at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
18:48:28        at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:755)
18:48:28        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:454)
18:48:28        at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:133)
18:48:28        at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
18:48:28        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
18:48:28        at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
18:48:28        at org.apache.spark.scheduler.Task.run(Task.scala:131)
18:48:28        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
18:48:28        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
18:48:28        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
18:48:28        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
18:48:28        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
18:48:28        ... 1 more
18:48:28    Caused by: java.io.FileNotFoundException: File not present on S3
18:48:28        at com.amazon.ws.emr.hadoop.fs.s3.S3FSInputStream.read(S3FSInputStream.java:273)
18:48:28        at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
18:48:28        at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
18:48:28        at java.io.FilterInputStream.read(FilterInputStream.java:83)
18:48:28        at org.apache.parquet.io.DelegatingSeekableInputStream.read(DelegatingSeekableInputStream.java:61)
18:48:28        at org.apache.parquet.bytes.BytesUtils.readIntLittleEndian(BytesUtils.java:80)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:528)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:513)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:507)
18:48:28        at org.apache.parquet.hadoop.ParquetFileReader.readFooter(ParquetFileReader.java:456)
18:48:28        at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat.getFooter(ParquetFileFormat.scala:644)
18:48:28        at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat.$anonfun$buildPrefetcherWithPartitionValues$1(ParquetFileFormat.scala:515)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader.org$apache$spark$sql$execution$datasources$AsyncFileDownloader$$downloadFile(AsyncFileDownloader.scala:93)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader$$anon$1.call(AsyncFileDownloader.scala:73)
18:48:28        at org.apache.spark.sql.execution.datasources.AsyncFileDownloader$$anon$1.call(AsyncFileDownloader.scala:72)
18:48:28        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
18:48:28        ... 3 more
```

### System information
**The output of `dbt --version`:**
```
Core:
  - installed: 1.3.2
  - latest:    1.4.1 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - glue:  1.3.12 - Up to date!
  - spark: 1.3.1  - Update available!

  At least one plugin is out of date or incompatible with dbt-core.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation
```

**The operating system you're using:**

**The output of `python --version`:**
```
Python 3.8.16
```

### Additional context
Initial run of model creation works fine - it breaks only when I re-run it the second time for incremental update!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 ## dbt-glue 1.0.0 (Release TBD)
 
+## v1.3.14
+- Fix HUDI merge strategy
+
 ## v1.3.13
 - Fix exception handling for correct DBT report
 - Update READEME.md to use Apache Iceberg Connector for AWS Glue v0.14.0

--- a/dbt/adapters/glue/impl.py
+++ b/dbt/adapters/glue/impl.py
@@ -646,7 +646,7 @@ PARTITIONED BY ({part_list})
             }
 
         if isTableExists:
-            write_mode = 'Overwrite'
+            write_mode = 'Append'
             write_operation_config = {
                 'hoodie.upsert.shuffle.parallelism': 20,
                 'hoodie.datasource.write.operation': 'upsert',
@@ -654,7 +654,7 @@ PARTITIONED BY ({part_list})
                 'hoodie.cleaner.commits.retained': 10,
             }
         else :
-            write_mode = 'Append'
+            write_mode = 'Overwrite'
             write_operation_config = {
                 'hoodie.bulkinsert.shuffle.parallelism': 20,
                 'hoodie.datasource.write.operation': 'bulk_insert',

